provisioning/miniprov/hare_mp/store.py

```diff
@@ -41,3 +50,20 @@ def __init__(self, url: str):
 
     def _raw_get(self, key: str) -> str:
         return self.conf.get('hare', key)
+
+    def get_machine_id(self) -> str:
+        with open('/etc/machine_id', 'r') as f:
+            machine_id = f.readline().strip('\n')
+            return machine_id
+        raise RuntimeError('cannot read machine_id')
```

Comment:
[Minor] this instruction looks not feasible (dead code). Can be easily removed.

Reply:
Yes. Otherwise `with` construct would have not much sense.

From [Python docs](https://docs.python.org/3/library/functions.html#open):
> Open file and return a corresponding file object. If the file cannot be opened, an OSError is raised. See Reading and Writing Files for more examples of how to use this function.